frontend/my-app/src/Basket.tsx

```diff
@@ -0,0 +1,194 @@
+            <div> Basket is empty!
+            <br/>
+                To add a product, click on it!
+            </div>
+            <button onClick={returnToMainPage}> Return to main page</button>
+            </>
+        )
+    }
+    type ProductDetails = {
+        amount: number;
+        unitPrice: string;
+        maxAmount?: number;
+        photo?: string
+    };
+
+    type BasketItem = [string, ProductDetails];
+    const basketArr: BasketItem[] = Object.entries(cookies.basketCookie) ? Object.entries(cookies.basketCookie) : []
+    const total = basketArr.reduce((acc, curr) => {
+        const [productName, productDetails] = curr;
+            const productTotal = productDetails.amount * parseFloat(productDetails.unitPrice.replace(',', '.'));
+            return acc + productTotal;
+
+    }, 0);
+
+    const sendOrderToDatabase  = async (e: FormEvent) => {
+        e.preventDefault()
+        if (user === '' || userAddress === ''){
+            alert('User name and address are mandatory!')
+            return;
+        }
+        interface tempObject {
+            basketCookie: {},
+            user: string,
+            address: string
+        }
+        const _tempObject: tempObject = {
+            basketCookie: {},
+            user: '',
+            address: ''
+        }
+        _tempObject.basketCookie = cookies.basketCookie
+        _tempObject.user = user
+        _tempObject.address = userAddress
+
+        const res = await fetch(`${apiUrl}/addOrder`, {
+            method: 'POST',
+            body: JSON.stringify(_tempObject),
+            headers: {
+                "Content-Type": "application/json"
+            },
+        });
+        setCookie('basketCookie', {})
+        alert('Thank you for your order!')
+        navigate('/')
+
+    };
+    const importedImages: any = {}
+    for (let i=0; i < basketArr.length; i++){
+        importedImages[basketArr[i][0]] = require(`./${basketArr[i][1].photo}`)
+    }
+
+    const increaseAmount = (key: string) => {
+
+        if (cookies.basketCookie[key].amount < Number(cookies.basketCookie[key].maxAmount)){
+        console.log(cookies.basketCookie[key].amount, Number(cookies.basketCookie[key].maxAmount))
+            const newAmount = cookies.basketCookie[key].amount + 1
+            const newCookie = cookies.basketCookie
+            newCookie[key].amount = newAmount
+            setCookie('basketCookie', newCookie)
+        } else {
+            alert('Not enough items in storage :(')
+        }
+    }
+
+    const decreaseAmount = (key: string) => {
+        if (cookies.basketCookie[key].amount > 0){
+            const newAmount = cookies.basketCookie[key].amount - 1
+            const newCookie = cookies.basketCookie
+            newCookie[key].amount = newAmount
+            setCookie('basketCookie', newCookie)
+        }
+
+    }
+
+    const divs = basketArr.map((arr) => {
+        if (Number(arr[1].amount) > 0) {
+            return <div key={arr[0]}

```

Comment:
this file seems to big. Couldn't, for example, this part be extracted into separate file?